addons/l10n_it_edi_withholding/__init__.py

```diff
@@ -0,0 +1,101 @@
+
+            # Increase the sequence number of the old taxes multiplying it by 10 and adding 21
+            # so that the withholding can have sequence=10 and the pension fund sequence=20
+            offset = 21
+            all_taxes = env['account.tax'].search([('company_id', '=', company.id)])
+            for tax in all_taxes.filtered(lambda x: (x.sequence <= 20 and not x.l10n_it_pension_fund_type and not x.l10n_it_withholding_type)):
+                if not tax.l10n_it_withholding_type and not tax.l10n_it_pension_fund_type:
+                    tax.sequence += offset
+            _logger.info("Increased sequence number of old taxes by %s for company: %s(%s).", offset, company.name, company.id)
+    except psycopg2.errors.UniqueViolation:
+        generated_taxes_ref = {}
+        _logger.error("Withholding and Pension Fund taxes already exist for company: %s(%s).", company.name, company.id)
+    except Exception:
+        generated_taxes_ref = {}
+        _logger.error("Can't load Withholding and Pension Fund taxes for company: %s(%s).", company.name, company.id)
+
+    return generated_taxes_ref
+
+def _l10n_it_edi_setup_accounts_on_taxes(env, company, generated_taxes_ref, generated_accounts_ref):
+    """ Setup the accounts on the taxes, after the accounts have been created """
+    try:
+        with env.cr.savepoint():
+            # Set the transition account
+            for tax, value in generated_taxes_ref['account_dict']['account.tax'].items():
+                transition_account = value['cash_basis_transition_account_id']
+                if transition_account:
+                    tax.cash_basis_transition_account_id = generated_accounts_ref.get(transition_account)
+
+            # The tax repartition lines accounts has already been generated from the template by l10n_it
+            referenced_accounts = {}
+            repartitions_dict = generated_taxes_ref['account_dict']['account.tax.repartition.line']
+            for dummy, value_dict in repartitions_dict.items():
+                account_template = value_dict['account_id']
+                template_xml_id = account_template.get_external_id()[account_template.id]
+                xml_id = template_xml_id.split('.')
+                referenced_accounts[template_xml_id] = env.ref(f'{xml_id[0]}.{company.id}_{xml_id[1]}')
+
+            # Set the tax repartition lines accounts on the generated taxes
+            for repartition_line, value in repartitions_dict.items():
+                template = value['account_id']
+                if template:
+                    repartition_line.account_id = referenced_accounts[template.get_external_id()[template.id]]
+            _logger.info("Cash Basis Transition Accounts and Repartition accounts on taxes for company: %s(%s).", company.name, company.id)
+    except Exception:
+        _logger.error("Cannot setup Cash Basis Transition Accounts and Repartition accounts on taxes for company: %s(%s).", company.name, company.id)
```

Comment:
```suggestion
```  Let it traceback and have the courage to admit your errors.

Reply:
I thought you said that I had to trap the errors. 🤷🏻